python/GafferUI/PresetsPlugValueWidget.py

```diff
@@ -120,11 +124,6 @@ def __menuDefinition( self ) :
 		# Required for context-sensitive dynamic presets
 		with self.getContext():
```

Comment:
All looks good to me.  The only thing I'm not sure I get:  we scope this context to allow for dynamic presets - that seems to imply that these these dynamic presets would involve computes, which could also cause blocking?  If so, this PR is still making things better, is this just a case where we want to improve the common case first?

Reply:
Yep, that's correct. To be honest, my main focus in this PR was on fixing the toolbar flicker, and the refactoring to store `self.__currentPreset` was mostly in aid of that. I hadn't considered that I also could have stored the full list of available presets, but now you point it out, that seems like it would be a good change. Let's get this out of the door first though...